File: src/org/newtonpolyhedron/solve/surface/SurfaceBuilder.java
package org.newtonpolyhedron.solve.surface;

import java.util.Map;

import org.fs.utils.collection.set.IndexedSet;
import org.fs.utils.collection.table.KeyTable;
import org.newtonpolyhedron.entity.Surface;
import org.newtonpolyhedron.entity.vector.IntVector;

public interface SurfaceBuilder {
	
	public Map <Integer, IndexedSet <Surface>> getSurfaces(KeyTable <IntVector, Integer, Boolean> lookupTable, int dim);
	
}

File: src/org/newtonpolyhedron/solve/surface/SurfaceBuilder.scala
package org.newtonpolyhedron.solve.surface

import scala.collection.immutable.SortedSet

import org.fs.utils.collection.table.KeyTable
import org.newtonpolyhedron.entity.Surface
import org.newtonpolyhedron.entity.vector.VectorImports._

trait SurfaceBuilder {
  def surfaces(lookupTable: KeyTable[IntVec, Int, Boolean], dim: Int): Map[Int, SortedSet[Surface]]
}